hs-bindgen/src-internal/HsBindgen/Frontend/AST/Internal.hs

```diff
@@ -181,9 +182,21 @@ data EnumConstant p = EnumConstant {
 data Function p = Function {
       functionArgs :: [Type p]
     , functionRes  :: Type p
+    , functionAttrs :: [FunctionAttr]
     , functionAnn  :: Ann "Function" p
     }
 
+-- | Function attributes specify function properties that may help the C
+-- compiler.
+--
+-- TODO: should this have a @p@ type argument as well?
```

Comment:
No need, unless or until we add something like a nested type or identifier in here.